python/src/opendp/_lib.py

```diff
@@ -26,40 +24,14 @@ def _load_library():
         lib_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), *['..'] * 3, 'rust', 'target', build_dir)
 
     if os.path.exists(lib_dir):
-        # Mapping of Python platform to library name format
-        platform_to_name_template = {
-            "darwin": "libopendp{}.dylib",
-            "linux": "libopendp{}.so",
-            "win32": "opendp{}.dll",
-        }
-        # Mapping of Python platform/machine to Rust architecture.
-        platform_machine_to_architecture = {
-            ("win32", "AMD64"): "x86_64",
-            # No way to build this yet ("win32", "ARM64"): "aarch64",
-            ("darwin", "x86_64"): "x86_64",
-            ("darwin", "arm64"): "aarch64",
-            ("linux", "x86_64"): "x86_64",
-            ("linux", "aarch64"): "aarch64",
-        }
-
-        name_template = platform_to_name_template.get(sys.platform)
-        if name_template is None:
-            raise Exception("Platform not supported", sys.platform)
-        architecture = platform_machine_to_architecture.get((sys.platform, platform.machine()))
-        if architecture is None:
-            raise Exception("Machine not supported", sys.platform, platform.machine())
-
-        def get_lib_path(name_template, architecture):
-            suffix = f"-{architecture}" if architecture is not None else ""
-            name = name_template.format(suffix)
-            return os.path.join(lib_dir, name)
-
-        # First try name with architecture
-        lib_path = get_lib_path(name_template, architecture)
-        if not os.path.exists(lib_path):
-            # Fall back to name without architecture (happens on darwin, which has fat binaries, and developer installs)
-            lib_path = get_lib_path(name_template, None)
-
+        def is_binary(name):
+            return any(name.endswith(ext) for ext in (".so", ".dylib", ".dll"))
+        
+        lib_dir_file_names = [n for n in os.listdir(lib_dir) if is_binary(n)]
+        if len(lib_dir_file_names) != 1:
+            raise Exception(f"Expected exactly one binary to be present. Got: {lib_dir_file_names}")
+        
+        lib_path = os.path.join(lib_dir, lib_dir_file_names[0])
```

Comment:
FWIW, pathlib is nice for cases like this:
```suggestion
        lib_path = Path(lib_dir) / lib_dir_file_names[0]
```
or if `lib_dir` is already a Path, just
```suggestion
        lib_path = lib_dir / lib_dir_file_names[0]
```